src/inmanta/data/dataview.py

```diff
@@ -1239,3 +1246,91 @@ def construct_dtos(self, records: Sequence[Record]) -> Sequence[dict[str, str]]:
             ).dict()
             for res in records
         ]
+
+
+class PreludeBasedFilteringQueryBuilder(SimpleQueryBuilder):
+    """
+    A query builder that applies any filters and the LIMIT statement to the prelude query rather than the outer query.
```

Comment:
```suggestion
    A query builder that applies any filters and the LIMIT statement to the prelude query rather than the outer query.
    The outer query may use the table name "prelude" to refer to the inner query.
```